app/helpers/application_helper.rb

```diff
@@ -148,13 +148,33 @@ def model_to_controller(record)
     record.class.base_model.name.underscore
   end
 
+  def controller_to_model_params
+    {
+      "ManageIQ::Providers::CloudManager::Template" => VmOrTemplate,
+      "ManageIQ::Providers::CloudManager::Vm"       => VmOrTemplate,
+      "ManageIQ::Providers::InfraManager::Template" => VmOrTemplate,
+      "ManageIQ::Providers::InfraManager::Vm"       => VmOrTemplate
+    }
+  end
+  private :controller_to_model_params
+
+  def model_string_to_constant_params
+    {
+      "all_vms" => VmOrTemplate
+    }
+  end
+  private :model_string_to_constant_params
+
+  def type_has_quadicon(type)
+    !["ManageIQ::Providers::Foreman::ConfigurationManager::ConfigurationProfile", "ServiceTemplate"].include? type
+  end
+
   def controller_to_model
-    case self.class.model.to_s
-    when "ManageIQ::Providers::CloudManager::Template", "ManageIQ::Providers::CloudManager::Vm", "ManageIQ::Providers::InfraManager::Template", "ManageIQ::Providers::InfraManager::Vm"
-      VmOrTemplate
-    else
-      self.class.model
-    end
+    controller_to_model_params[self.class.model.to_s] || self.class.model
```

Comment:
Believe it or not, I thought the original code was easier to understand.  However, perhaps it might be clearer with something like...?

```ruby
CONTROLLER_TO_MODEL = Hash.new { |_h, k| k.safe_constantize }.merge(
  "ManageIQ::Providers::CloudManager::Template" => VmOrTemplate,
  "ManageIQ::Providers::CloudManager::Vm"       => VmOrTemplate,
  "ManageIQ::Providers::InfraManager::Template" => VmOrTemplate,
  "ManageIQ::Providers::InfraManager::Vm"       => VmOrTemplate
).freeze

def controller_to_model
  CONTROLLER_TO_MODEL[self.class.model.name]
end
```

Not sure.  I really don't like the hardcoded models, particular that they are hardcoded as strings, but I am not sure any other way to do it at the moment.